pkg/nmi/server/server.go

```diff
@@ -463,17 +491,18 @@ func listPodIDsWithRetry(ctx context.Context, kubeClient k8s.Client, logger *log
 				for _, podID := range idStateMap[""] {
 					if strings.EqualFold(rqClientID, podID.Spec.ClientID) {
 						logger.Warningf("found assignedIDs with no state for pod:%s/%s. AssignedIDs created with old version of mic.", podns, podname)
-						return idStateMap[""], nil
+						return idStateMap[""], true, nil
 					}
 				}
 				for _, podID := range idStateMap[aadpodid.AssignedIDAssigned] {
 					if strings.EqualFold(rqClientID, podID.Spec.ClientID) {
-						return idStateMap[aadpodid.AssignedIDAssigned], nil
+						return idStateMap[aadpodid.AssignedIDAssigned], true, nil
```

Comment:
When do we return identities which are in created state ?

Reply:
We don't need to return identities which are in created state. We only need identities in Assigned state.